scripts/o3de/o3de/engine_template.py

```diff
@@ -2119,6 +2121,15 @@ def create_gem(gem_path: pathlib.Path,
     # remove the first and last quote because those already exist in gem.json
     replacements.append(("${UserTags}", tags_quoted[1:-1]))
 
+    temp_platforms = []
+    if platforms:
+        #same as tags, allow commas or spaces as platform separators
+        new_platforms = platforms.replace(',', ' ').split() if isinstance(platforms, str) else platforms
+        temp_platforms.extend(new_platforms)
```

Comment:
`gem_properties` has a function called `update_values_in_key_list` - if it can be used here - can we move that function into `utils.py` and use it everywhere we are modifying these kind of quoted lists?